include/service/entities/custom_object_service.js

```diff
@@ -955,7 +955,7 @@ module.exports = function CustomObjectServiceModule(pb) {
             options = {};
         }
 
-        if (!pb.validation.isIdStr(id, true)) {
+        if (!pb.validation.isIdStr(id.toString(), true)) {
```

Comment:
Ok. Then I was totally wrong in my conversation with @amirsaber today.
On Nov 16, 2015 5:44 PM, "Brian" <notifications@github.com> wrote:

> In include/service/entities/custom_object_service.js
> <https://github.com/pencilblue/pencilblue/pull/861#discussion_r44996210>:
>
> > @@ -955,7 +955,7 @@ module.exports = function CustomObjectServiceModule(pb) {
> >              options = {};
> >          }
> >
> > -        if (!pb.validation.isIdStr(id, true)) {
> > +        if (!pb.validation.isIdStr(id.toString(), true)) {
>
> That is incorrect. The ValidationService.isId function checks both
> Strings and ObjectID objects. ValidationService.isIdStr adds the
> additional constraint that the value must also be a string as well as
> parseable into an ObjectID object. If you look at the implementation of the
> isIdStr function you'll see that it validates the value as a string then
> calls the isId function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pencilblue/pencilblue/pull/861/files#r44996210>.
>